lib/hysync/marc_synchronizer/runner.rb

```diff
@@ -30,18 +29,17 @@ def run(force_update = false)
 
       # Adds a collection term to the given base_digital_object_data if
       # collection info is available from the given marc_record.
-      def add_collection_if_773_w_clio_id_present!(digital_object_data, marc_record)
+      # @param digital_object_data [Hash] Hyacinth digital object properties
+      # @param marc_record [MARC::Reader] ruby-marc record object
+      def add_collection_if_collection_clio_id_present!(digital_object_data, marc_record)
         # If this marc record has a collection clio id, create or retrieve the
         # controlled term associated with that ID. This collection clio id, if present,
         # will be in 773 $w and will be a 7 or 8 digit number that's optionally prefixed
         #  with "(NNC)".  The referenced clio record must also be a collection-level record,
         # as indicated by MARC leader byte 7 having a value of'c'.
-        collection_field_773 = MarcSelector.first(marc_record, 773, w: true)
-        if collection_field_773
-          match = collection_field_773['w'].match(/^(\(NNC\))*(\d{7,8})$/)
-          # If 773 $w doesn't match our record, this isn't a valid CLIO id reference
-          return unless match
-          collection_clio_id = match[2] # retrieve numeric portion of ID
+        digital_object_data['dynamic_field_data'].fetch("collection", []).each do |collection_term|
+          collection_clio_id = collection_term["clio_id"]
+          next unless collection_clio_id
           unless @collection_clio_ids_to_uris.key?(collection_clio_id)
             collection_marc_record = @voyager_client.find_by_bib_id(collection_clio_id)
```

Comment:
We should log an error (by pushing a message onto the @errors array) if an associated marc record isn't found for this CLIO id, or if a marc record is found that doesn't have leader[7] == 'c'.  The digital object creation/update will fail with your new implementation (when it used to just omit the collection), and that seems like a fine new behavior since you'll be creating digital_object_data with an invalid term -- one that only includes a clio_id property, when 'value' or 'uri' are minimally required for a valid term -- but I think it would be useful for troubleshooting if a detailed error was explicitly added here.  Something like "For bib record 12345, could not resolve collection clio id #{collection_clio_id} to a collection-level marc record."

Reply:
do you want that in this PR or in a separate commit?

Reply:
Is PR okay?

Reply:
sure!

Reply:
See line 45-49 below

Reply:
Thanks!